controls/def.cf

```diff
@@ -646,10 +647,15 @@ bundle common inventory_control
       "disable_inventory_lsb" expression => "disable_inventory";
       "disable_inventory_lsb" not => fileexists($(lsb_exec));
 
+      # If we have /proc/device-tree we should likely disable dmi completely
+      # as of 2022 systems with dmi dont have device-tree and vice versa.
+      "have_proc_device_tree" expression => fileexists($(proc_device_tree));
+
       # by default disable the dmidecode inventory if the general
       # inventory is disabled or the binary does not exist.  Note that
       # typically this is a very fast binary.
       "disable_inventory_dmidecode" expression => "disable_inventory";
+      "disable_inventory_dmidecode" expression => "have_proc_device_tree";
```

Comment:
I don't think this does what you perhaps hope it does.

This class protects the actuation of the bundle after the agent has started executing the bundle sequence, but that does not necessarily protect against vars and classes resolution in `bundle agent cfe_autorun_inventory_dmidecode`. All of those inventory autorun bundles are affected in the same way, defiing the class isn't necessarily going to prevent them from resolving those things, but they will have less opportunity.

To really disable it, those bundles need to be treated and have guards against resolving variables when the class is defined.
Really annoying, and I am not suggesting that you not do this or that you go the extra mile, just highlighting the fact.

Reply:
No, that makes sense. vars/classes are different. ack.

Reply:
Taking a further look I think I'll skip a bigger refactor. Let me know and we can just remove this line as it works fine without the inhibition as well.